crates/ruff_python_formatter/src/snapshots/ruff_python_formatter__tests__ruff_test__while_py.snap

```diff
@@ -0,0 +1,70 @@
+---
+source: crates/ruff_python_formatter/src/lib.rs
+expression: snapshot
+---
+## Input
+```py
+while 34: # trailing test comment
+    pass # trailing last statement comment
+
+    # trailing while body comment
+
+# leading else comment
+
+else: # trailing else comment
+    pass
+
+    # trailing else body comment
+
+
+while aVeryLongConditionThatSpillsOverToTheNextLineBecauseItIsExtremellyLongAndGoesOnAndOnAndOnAndOnAndOnAndOnAndOnAndOnAndOn: # trailing comment
+    pass
+
+else:
+    ...
+
+while (
+    some_condition(unformatted, args) and anotherCondition or aThirdCondition
+):  # comment
+    print("Do something")
+
+
+while (
+    some_condition(unformatted, args) # trailing some condition
+    and anotherCondition or aThirdCondition  # trailing third condition
+):  # comment
+    print("Do something")
+```
+
+
+
+## Output
+```py
+while 34:  # trailing test comment
+    pass  # trailing last statement comment
+
+    # trailing while body comment
+
+# leading else comment
+
+else:  # trailing else comment
+    pass
+
+    # trailing else body comment
+while (
```

Comment:
Should this newline be removed?

Reply:
Good catch. No it should not. @konstin is working on keeping the right amount of newlines between top level statements in https://github.com/charliermarsh/ruff/pull/4808 I should probably rebase on top of that.